CalPatRec/src/HelixFinder_module.cc

```diff
@@ -0,0 +1,2559 @@
+          }
+          int hitIndice = _tcHits[q].hitIndice;
+          nStrawHitsInTimeCluster = nStrawHitsInTimeCluster + _chColl->at(hitIndice).nStrawHits();
+          nComboHitsInTimeCluster = nComboHitsInTimeCluster + 1;
+          if (_tcHits[q].used == false) {
+            continue;
+          }
+          nStrawHitsInHelix = nStrawHitsInHelix + _chColl->at(hitIndice).nStrawHits();
+          nComboHitsInHelix = nComboHitsInHelix + 1;
+        }
+        if (nStrawHitsInHelix >= _minNHelixStrawHits && nComboHitsInHelix >= _minNHelixComboHits) {
+          if (_debug == 1 && _runDisplay == 1) {
+            trackingInfo.finalNHits = _circleFitter.qn();
+            trackingInfo.becameHelix = true;
+            _plottingData.push_back(trackingInfo);
+          }
+          saveHelixCandidate();
+          saveHelix(tc, *_hsColl);
+          if (_diagLevel == 1) {
+            _diagInfo.nHelices++;
+          }
+          int remainingStrawHits = nStrawHitsInTimeCluster - nStrawHitsInHelix;
+          int remainingComboHits = nComboHitsInTimeCluster - nComboHitsInHelix;
+          if (remainingStrawHits < _minNHelixStrawHits ||
+              remainingComboHits < _minNHelixComboHits) {
+            i = _tcHits.size();
+            j = i;
+            break;
+          } else {
+            findAnotherHelix = true;
+            if (_doIsolationFlag == true || _doAverageFlag == true || _doLineFlag == true) {
+              resetFlags();
+            }
+            i = _tcHits.size();
+            j = i;
+            break;
+          }
+        } else {
+          if (_debug == 1 && _runDisplay == 1) {
+            trackingInfo.finalNHits = _circleFitter.qn();
+            trackingInfo.becameHelix = false;
+            _plottingData.push_back(trackingInfo);
+          }
+        }
+      }
+    }
+    _tcHits[i].uselessTripletSeed = uselessSeed;
+  }
+}
+
+//-----------------------------------------------------------------------------
+// check flags to see if point is good for triplet-ing with
+//-----------------------------------------------------------------------------
+bool HelixFinder::passesFlags(size_t& tcHitsIndex) {
+
+  if (_tcHits[tcHitsIndex].inHelix == true) {
+    return false;
+  }
+  if (_tcHits[tcHitsIndex].uselessTripletSeed == true) {
+    return false;
+  }
+  if (_doIsolationFlag == true && _tcHits[tcHitsIndex].isolated == true) {
+    return false;
+  }
+  if (_doAverageFlag == true && _tcHits[tcHitsIndex].averagedOut == true) {
+    return false;
+  }
+  if (_doLineFlag == true && _tcHits[tcHitsIndex].notOnLine == true) {
+    return false;
+  }
+
+  return true;
+}
+
+//-----------------------------------------------------------------------------
+// set ith point of triplet, return outcome value which can be used to direct for loops
+//-----------------------------------------------------------------------------
+void HelixFinder::setTripletI(size_t& tcHitsIndex, triplet& trip, int& outcome) {
```

Comment:
Please use an enum instead of int fo outcome.  If it is only 0 or -1 then umse a bool.